ls1intum/Artemis
# Enhancement/programming exercise/dynamically build build plan links

### Checklist

### Motivation and Context


Fixes #1184 

We currently have hardcode the links to the build plans of programming exercises in all components. This is far from optimal and especially problematic with the new Jenkins CI alternative.

### Description

I added a new directive `build-plan-link.directive.ts` that adds the appropriate attributes to its host component. The link gets created dynamically based on the value provided by the profile service, which in turn gets the link template from the call to the `/info` actuator endpoint.

### Steps for Testing


1. Log in to Artemis
2. Check the build plan links on the following pages
3. Programming exercise details
4. Course overvice
5. Exercise scores page
6. Exercise participation list
